File: implementations/node/sdk/src/contracts/control/execution-cancellation-response.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionStatus } from "../executions/execution-status.js";

export interface AiSdkExecutionCancellationResponse {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionCancellationResponse;
  readonly executionId: string;
  readonly cancellationRequested: boolean;
  readonly status: AiSdkExecutionStatus;
  readonly requestedAtUtc?: string;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/control/execution_cancellation_response.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from ..executions.execution_status import AiSdkExecutionStatus


@dataclass(frozen=True)
class AiSdkExecutionCancellationResponse(AiSdkWireModel):
    execution_id: str
    cancellation_requested: bool
    status: AiSdkExecutionStatus
    schema_version: int = AiSdkSchemaVersions.EXECUTION_CANCELLATION_RESPONSE
    requested_at_utc: str | None = None
    correlation_id: str | None = None
